monitor_func.go

```diff
@@ -6,7 +6,7 @@ import (
 )
 
 // MonitorFunc turns a function into a Monitor
-func MonitorFunc(interval time.Duration, logFunc func(Stats)) Monitor {
+func MonitorFunc(interval time.Duration, logFunc func(Stats)) *monitorFunc {
```

Comment:
I would not not return an unexported value (see my previous comment).

Reply:
same